File: src/main/UserService.java
package main;

import java.time.LocalDateTime;
import java.util.HashSet;

public class UserService {
    private Users user;

    public Users getUser() {
        return user;
    }

    public void setUser(Users user) {
        this.user = user;
    }

    public QuizDetails joinQuiz(Quizs quiz) {
        QuizDetails quizDetail = new QuizDetails();
        quizDetail.setUser(user);
        quizDetail.setQuiz(quiz);
        quizDetail.setStartedAt(LocalDateTime.now());

        if(user.getQuizDetails() == null) {
            user.setQuizDetails(new HashSet<>());
        }

        user.getQuizDetails().add(quizDetail);
        return quizDetail;
    }

    public int submitQuiz(QuizDetails quizDetails) {
        QuizService quizService = new QuizService();
        quizService.setQuiz(quizDetails.getQuiz());
        int totalScore = 0;
        LocalDateTime timeEndQuiz = quizDetails.getStartedAt().plusMinutes(quizDetails.getQuiz().getDuration());
        for(UserAnswers userAnswer : quizDetails.getUserAnswers()) {
            if(userAnswer.getSubmittedAt().isBefore(timeEndQuiz) && quizService.checkUserAnswerInQuiz(userAnswer)) {
                totalScore ++;
            }
        }
        quizDetails.setScore(totalScore);
        return totalScore;
    }
}

File: src/QuizAppCSharp/QuizApp/Service/UserService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuizApp.Service
{
    public class UserService
    {
        public User User { get; set; }
        public UserService() { 
        
        }

        public QuizDetail JoinQuiz(Quiz quiz)
        {
            QuizDetail quizDetail = new QuizDetail();
            quizDetail.User = User;
            quizDetail.Quiz = quiz;
            quizDetail.StartedAt = DateTime.Now;

            if(User.QuizDetails == null)
            {
                User.QuizDetails = new HashSet<QuizDetail>();
            }

            User.QuizDetails.Add(quizDetail);
            return quizDetail;
        }

        public int SubmitQuiz(QuizDetail quizDetail)
        {
            QuizService quizService = new QuizService();
            quizService.Quiz = quizDetail.Quiz;
            int totalScore = 0;
            DateTime timeEndQuiz = quizDetail.StartedAt.AddMinutes(quizDetail.Quiz.Duration);
            foreach(UserAnswer userAnswer in quizDetail.UserAnswers) {
                if (userAnswer.SubmittedAt <= timeEndQuiz && quizService.CheckUserAnswerInQuiz(userAnswer))
                {
                    totalScore++;
                }
            }
            quizDetail.Score = totalScore;
            return totalScore;
        }
    }
}
